test/e2e/common/node/container_probe.go

```diff
@@ -95,7 +95,7 @@ var _ = SIGDescribe("Probing container", func() {
 		}
 
 		restartCount := getRestartCount(p)
-		framework.ExpectEqual(restartCount, 0, "pod should have a restart count of 0 but got %v", restartCount)
+		gomega.Expect(restartCount).To(gomega.BeIdenticalTo(0), "pod should have a restart count of 0 but got %v", restartCount)
```

Comment:
I'm curious as to the reasoning behind the `BeIdenticalTo` usage here - Non-standard `Equal` checks kinda draw attention to themselves imo?

Reply:
@endocrimes 

In this case, their functions are the same. The only difference is that BeIdenticalTo uses the `==` to compare, while Equal uses `reflect.DeepEqual` to compare. Relatively speaking, the performance of BeIdenticalTo may be better. Of course, if this is not considered, Equal is also reasonable for a unified style.

Reply:
I understand the behavior difference :) - My question was more about the reason behind using them in this particular case, when not trying to test for pointer equality explicitly 😅   - Definitely would rather stick to a unified style and stick to using exceptions only when they're needed